### PR TITLE
docs: change `line_conditional_grid_dash` to even more advanced `line_conditional_axis`

### PR DESCRIPTION
**Pull request by kanitw**
also simplify its predicate

### CODE REVIEW

**Comment by kanitw** [comment]
Hmm, the hack to use whitespace instead of `labelOffset` doesn't work for SVG generation. (It worked OK on Chrome.)

File: site/docs/encoding/axis.md [hunk 1]
```diff
@@ -117,7 +117,9 @@ Note that each axis tick, grid line, and label instance is backed by a data obje
 - `label` - the string label
 - `value` - the data value
 
-For example, we can adjust the grid dash in a line chart based on whether a particular grid line falls on a year boundary:
+### Example: Conditional Axis Properties and Multi-Line Axis Label
+
+In the following example, we adjust the `gridDash` and `tickDash` properties in a line chart based on whether a particular grid line falls on a year boundary. We also use the `labelExpr` property to show multi-line labels for year and month, showing the year number only for January of each year.
 
 <div class="vl-example" data-name="line_conditional_grid_dash"></div>
```
Comment (domoritz) [inline site/docs/encoding/axis.md]: Please rename references as well. 

```suggestion
<div class="vl-example" data-name="line_conditional_axis_"></div>
```

**Comment by domoritz** [comment]
I think we can merge this despite bad offset btw.

**Comment by kanitw** [comment]
> I think we can merge this despite bad offset btw.

Why? It's bad. I'd rather wait to fix the offset.

**Comment by domoritz** [comment]
Okay. Let’s wait then.

**Comment by kanitw** [comment]
This is rebased, now that we support `labelOffset`.

**Comment by domoritz** [comment]
> Hmm, the hack to use whitespace instead of labelOffset doesn't work for SVG generation. (It worked OK on Chrome.)

Vega now always trims text so the hack does not work anymore.

**Comment by kanitw** [comment]
line_conditional_axis_config.vl.json isn't correct yet, but it should be correct with Vega 5.10.  I'll track that in https://github.com/vega/vega-lite/pull/6033

**Pull request merged**